include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -31,22 +30,92 @@ namespace impl
+
+    template<class AssetType>
+    struct AssetBundleIterator
+    {
+        using iterator_category = std::random_access_iterator_tag;
+        using difference_type = std::ptrdiff_t;
+
+        AssetBundleIterator(const core::smart_refctd_ptr<asset::IAsset>* _ptr) : ptr(_ptr) {}
+
+        // general operators
+        inline AssetType* operator*()
+        {
+            return static_cast<AssetType*>(ptr->get());
+        }
+        inline const AssetType* operator*() const
+        {
+            return static_cast<const AssetType*>(ptr->get());
+        }
+        inline const core::smart_refctd_ptr<asset::IAsset>* operator->() const
+        {
+            return ptr;
+        }
+
+        // arithmetic operators
+        inline AssetBundleIterator<AssetType>& operator++()
+        {
+            ++ptr;
+            return *this;
+        }
+        inline AssetBundleIterator<AssetType> operator++(int)
+        {
+            return AssetBundleIterator<AssetType>(ptr++);
+        }
+        inline difference_type operator-(const AssetBundleIterator<AssetType>& other) const
+        {
+            return ptr - other.ptr;
+        }
+
+        // comparison operators
+        inline bool operator!=(const AssetBundleIterator<AssetType>& other) const
+        {
+            return ptr != other.ptr;
+        }
+
+    private:
+        const core::smart_refctd_ptr<asset::IAsset>* ptr;
+    };
 }
 
 
 
 class IGPUObjectFromAssetConverter
 {
+    private:
+
     public:
         struct SParams
         {
+            //! Required not null
+            ILogicalDevice* device = nullptr;
+            //! Required not null
+            asset::IAssetManager* assetManager = nullptr;
             IGPUPipelineCache* pipelineCache = nullptr;
+
+            IPhysicalDevice::SLimits limits;
+
+            // @sadiuk put here more parameters if needed
+
+            //! Queue used for transfer commands (data copies, etc.)
+            IGPUQueue* transferQueue = nullptr;
+            //! Queue used for compute commands (compute shader mip map generation, etc.)
+            IGPUQueue* computeQueue = nullptr;
+
+            //! If not null, fence will be written here. Written fence will be signaled once last operations of corresponding type (transfer/compute) are finished.
+            core::smart_refctd_ptr<IGPUFence>* const transferFence = nullptr;
+            core::smart_refctd_ptr<IGPUFence>* const computeFence = nullptr;
+
+            //! If not null, semaphore will be written here. Written semaphore will be signaled once last operations of corresponding type (transfer/compute) are finished.
+            core::smart_refctd_ptr<IGPUSemaphore>* const transferSemaphore = nullptr;
+            core::smart_refctd_ptr<IGPUSemaphore>* const computeSemaphore = nullptr;
+
+            core::smart_refctd_ptr<IGPUEvent>* const transferEvent = nullptr;
+            core::smart_refctd_ptr<IGPUEvent>* const computeEvent = nullptr;
```

Comment:
maybe make a per-queue struct and an enum
```cpp
enum E_QUEUE_USAGE
{
   EQU_TRANSFER,
   EQU_COMPUTE,
   EQU_COUNT
};
````

then it can be an array of `[EQU_COUNT]`